core/module.go

```diff
@@ -66,6 +66,32 @@ type Module struct {
 	InstanceID *call.ID
 }
 
+type SDKConfig struct {
+	Source string            `field:"true" name:"source" doc:"Source of the SDK. Either a name of a builtin SDK or a module source ref string pointing to the SDK's implementation."`
+	Pin    string            `field:"true" name:"pin" doc:"Commit sha pinned for the SDK"`
```

Comment:
I don't see the pin being used in this PR - so, I'm happy to remove it for now (so we don't block the important work), and we can add it back in later. Better to not have the field, than to have it partially implemented and confusing users.